### ISSUE TITLE
run_ruby_end2end_tests.sh fails because GRPC_LINUX_EPOLL is not defined

### ISSUE THREAD

**Issue by adelez** [grpc/grpc#13681]
https://sponge.corp.google.com/target?id=89163d94-7656-45e2-80ab-e5e9ee81d181&target=github/grpc/ruby_macos_dbg_native&searchFor=&show=FAILED&sortBy=STATUS

**Comment by apolcyn** [comment]
I believe this was just fixed by https://github.com/grpc/grpc/pull/13670

It looks like the `Skipping epollex becuase GRPC_LINUX_EPOLL is not defined.` are only benign log messages

The actual failure is in:

```
E1206 17:17:16.626395000 123145305526272 sync_posix.cc:47]             assertion failed: pthread_mutex_lock(mu) == 0
tools/run_tests/helper_scripts/run_ruby_end2end_tests.sh: line 29: 45189 Abort trap: 6           ruby src/ruby/end2end/multiple_killed_watching_threads_driver.rb
+ EXIT_CODE=1
```

But that `src/ruby/end2end/multiple_killed_watching_threads_driver.rb` is the test which was hitting a race condition that got fixed in https://github.com/grpc/grpc/pull/13670, so I think this is from an old run.

Leaving this open in case it's still seen after https://github.com/grpc/grpc/pull/13670
 has been picked up, but will close if it doesn't continue

**Comment by apolcyn** [comment]
should be fixed by #13760
